lightning/src/chain/keysinterface.rs

```diff
@@ -406,8 +407,28 @@ pub trait KeysInterface {
 
 	/// Get secret key material as bytes for use in encrypting and decrypting inbound payment data.
 	///
+	/// If the implementor of this trait supports [phantom node payments], then every node that is
+	/// intended to be included in the phantom invoice(s) route hints must return the same value from
+	/// this method.
+	///
 	/// This method must return the same value each time it is called.
+	///
+	///[phantom node payments]: crate::ln::channelmanager::ChannelManager::get_phantom_scid
 	fn get_inbound_payment_key_material(&self) -> KeyMaterial;
+
+	/// Get the phantom node secret key for use in receiving a [phantom node payment].
+	///
+	/// Must return the same value for all `scid`s in a given phantom invoice's route hints.
```

Comment:
This is super duper confusing to me. I think you mean "when an invoice is generated with multiple nodes' phantom SCID values, this must return the same key across each node. Note that if multiple phantom nodes are being operated by using different SCIDs, this may return different values for the later, different, SCID sets which are included in different invoices"? That's super wordy but as-is I have no idea what's going on.